src/controllers/mentionEachUser.ts

```diff
@@ -6,27 +6,53 @@ import { env } from "../typeDefinitions/default.types";
 import {
   UserArray,
   MentionEachUserOptions,
+  DevFlag,
 } from "../typeDefinitions/filterUsersByRole";
+import { mentionEachUserInMessage } from "../utils/guildRole";
 import { checkDisplayType } from "../utils/checkDisplayType";
 
 export async function mentionEachUser(
   transformedArgument: {
     roleToBeTaggedObj: MentionEachUserOptions;
     displayMessageObj?: MentionEachUserOptions;
+    channelId: number;
+    dev?: DevFlag;
   },
-  env: env
+  env: env,
+  ctx: ExecutionContext
 ) {
   const getMembersInServerResponse = await getMembersInServer(env);
   const roleId = transformedArgument.roleToBeTaggedObj.value;
   const msgToBeSent = transformedArgument?.displayMessageObj?.value;
+  const dev = transformedArgument?.dev?.value || false;
   // optional chaining here only because display message obj is optional argument
   const usersWithMatchingRole = filterUserByRoles(
     getMembersInServerResponse as UserArray[],
     roleId
   );
-  const responseData = checkDisplayType({
+  const payload = {
+    channelId: transformedArgument.channelId,
+    roleId: roleId,
+    message: msgToBeSent,
     usersWithMatchingRole,
-    msgToBeSent,
-  });
-  return discordTextResponse(responseData);
+  };
+  if (!dev || usersWithMatchingRole.length === 0) {
```

Comment:
how are users with roles fetched

Reply:
We fetch all the discord users first. This is something that I haven't changed. I've just diverged the flow based on dev flag.